spec/jobs/teachers/schedule_trs_sync_job_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "rails_helper"
+
+RSpec.describe Teachers::ScheduleTRSSyncJob, type: :job do
+  describe "#perform" do
+    let!(:teacher1) { FactoryBot.create(:teacher, trs_data_last_refreshed_at: 5.days.ago) }
+    let!(:teacher2) { FactoryBot.create(:teacher, trs_data_last_refreshed_at: 4.days.ago) }
+    let!(:teacher3) { FactoryBot.create(:teacher, trs_data_last_refreshed_at: 3.days.ago) }
+    let!(:teacher4) { FactoryBot.create(:teacher, trs_data_last_refreshed_at: 2.days.ago) }
+    let!(:teacher5) { FactoryBot.create(:teacher, trs_data_last_refreshed_at: 1.day.ago) }
+
+    it "schedules sync jobs for teachers ordered by trs_data_last_refreshed_at" do
+      ordered_teachers = Teacher.order(trs_data_last_refreshed_at: :asc).limit(described_class::BATCH_SIZE)
```

Comment:
Does defining an array of the teachers and seconds in the order you expect work? Rather than repeat the same code from job.

Reply:
I completely agree about keeping it dumb and not having logic in the tests; I was suggesting this. What do you think?

```ruby
    it "schedules sync jobs for teachers ordered by trs_data_last_refreshed_at" do
      [
        [teacher1, 0],
        [teacher2, 3],
        [teacher3, 6],
        [teacher4, 9],
        [teacher5, 12],
      ].each do |teacher, wait_time|
        allow(Teachers::SyncTeacherWithTRSJob).to receive(:set).with(wait: wait_time.seconds).and_return(Teachers::SyncTeacherWithTRSJob)

        expect(Teachers::SyncTeacherWithTRSJob).to receive(:perform_later).with(teacher:)
      end      

      described_class.perform_now
    end
```